src/Makefile.am

```diff
@@ -943,84 +943,7 @@ libbitcoinkernel_la_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS) -fvisibility=default
 # TODO: libbitcoinkernel is a work in progress consensus engine library, as more
 #       and more modules are decoupled from the consensus engine, this list will
 #       shrink to only those which are absolutely necessary.
-libbitcoinkernel_la_SOURCES = \
-  kernel/bitcoinkernel.cpp \
-  arith_uint256.cpp \
-  chain.cpp \
-  clientversion.cpp \
-  coins.cpp \
-  compressor.cpp \
-  consensus/merkle.cpp \
-  consensus/tx_check.cpp \
-  consensus/tx_verify.cpp \
-  core_read.cpp \
-  dbwrapper.cpp \
-  deploymentinfo.cpp \
-  deploymentstatus.cpp \
-  flatfile.cpp \
-  hash.cpp \
-  kernel/chain.cpp \
-  kernel/checks.cpp \
-  kernel/chainparams.cpp \
-  kernel/coinstats.cpp \
-  kernel/context.cpp \
-  kernel/cs_main.cpp \
-  kernel/disconnected_transactions.cpp \
-  kernel/mempool_persist.cpp \
-  kernel/mempool_removal_reason.cpp \
-  key.cpp \
-  logging.cpp \
-  node/blockstorage.cpp \
-  node/chainstate.cpp \
-  node/utxo_snapshot.cpp \
-  policy/feerate.cpp \
-  policy/fees.cpp \
-  policy/packages.cpp \
-  policy/policy.cpp \
-  policy/rbf.cpp \
-  policy/settings.cpp \
-  pow.cpp \
-  primitives/block.cpp \
-  primitives/transaction.cpp \
-  pubkey.cpp \
-  random.cpp \
-  randomenv.cpp \
-  scheduler.cpp \
-  script/interpreter.cpp \
-  script/script.cpp \
-  script/script_error.cpp \
-  script/sigcache.cpp \
-  script/solver.cpp \
-  signet.cpp \
-  streams.cpp \
-  support/lockedpool.cpp \
-  sync.cpp \
-  txdb.cpp \
-  txmempool.cpp \
-  uint256.cpp \
-  util/batchpriority.cpp \
-  util/chaintype.cpp \
-  util/check.cpp \
-  util/exception.cpp \
-  util/fs.cpp \
-  util/fs_helpers.cpp \
-  util/getuniquepath.cpp \
-  util/hasher.cpp \
-  util/moneystr.cpp \
-  util/rbf.cpp \
-  util/serfloat.cpp \
-  util/signalinterrupt.cpp \
-  util/strencodings.cpp \
-  util/string.cpp \
-  util/syserror.cpp \
-  util/thread.cpp \
-  util/threadnames.cpp \
-  util/time.cpp \
-  util/tokenpipe.cpp \
-  validation.cpp \
-  validationinterface.cpp \
-  versionbits.cpp \
-  warnings.cpp
+libbitcoinkernel_la_SOURCES = kernel/bitcoinkernel.cpp $(libbitcoin_kernel_a_SOURCES) $(libbitcoin_util_a_SOURCES) $(libbitcoin_consensus_a_SOURCES)
```

Comment:
In commit "kernel: De-duplicate source file list" (de6d79847537cd7d0dfc3e2dd7a57b39b4281b5a)

Can code comment above be updated to explain why util and consensus sources are added to kernel library sources, instead of the kernel library just depending on these libraries?

Reply:
I did not wan to change the way `libbitcoinkernel_la` is compiled in this pull request. The way it is now, it just follows the same pattern of `libbitcoinconsensus_la` when it comes to defining its sources. So there should be no change in linking behavior of the external `libbitcoinkernel` arising from this patchset. Besides, linking against the static internal libraries probably won't work anyway.